source/documentation/troubleshooting/line-endings.md

```diff
@@ -0,0 +1,23 @@
+## Push failure from Windows due to CRLF
+
+Pushing an app can fail if you are pushing from a Windows machine that has saved the app files with Windows-style (CRLF) line breaks.
+
+In interpreted languages, you will get a `No such file or directory` error.
+
+This is an example of the error with Ruby:
+
+```
+2016-12-02T06:03:25.95-0800 [APP/PROC/WEB/0]ERR /usr/bin/env: ruby
+2016-12-02T06:03:25.95-0800 [APP/PROC/WEB/0]ERR : No such file or directory
+2016-12-02T06:03:25.95-0800 [APP/PROC/WEB/0]OUT Exit status 127
+```
+
+If you find that pushing an app from a Windows machine is failing like this, you must make sure that files are not being saved with CRLF line breaks. Check the settings in your text editor or IDE.
```

Comment:
This is not trivial. CRLF are normal line endings on win. It's like asking linux/mac to do CRFL instead of LF...
We could start a flame which ending is correct, some printers also require CRLF for postscript documents.

Moreover, all those interpreted languages, when run on windows directly, work fine with windows line endings...

I wonder if the real issue had been some editor/svn/git adding one more extra line break.